File: 1544-count-good-nodes-in-binary-tree/1544-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        res = [0]
        def check(node, m):
            if not node:
                return
            if node.val >= m:
                res[0] += 1
            m = max(m, node.val)
            check(node.left, m)
            check(node.right, m)
        check(root, float('-inf'))
        return res[0]

File: 1544-count-good-nodes-in-binary-tree/1544-count-good-nodes-in-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var goodNodes = function(root) {
    let res = 0;
    const check = (node, m) => {
        if (!node) return
        if (node.val >= m) res++;
        m = Math.max(m, node.val);
        check(node.left, m);
        check(node.right, m);
    }
    check(root, -Infinity);
    return res;
};